### PR TITLE
add WishList page with functionality

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add design and functionality for the wishlist page

In this PR, **373 additions and 48 deletions** lines of code
- Some files contain only 5 to 6 lines of code
- You can ignore to review CSS and backend related files

### Please review this PR, and give your feedback
---

To Log In, you can use this email
```
email: johndoe@gmail.com
password: johnDoe123
```

### CODE REVIEW

**Review by VLeads — APPROVED** [review]
Nothing to suggest. Appreciate your code structure and coding practices. nice work !!

**Review by b0llu — APPROVED** [review]
Great work! Clean code nothing to suggest

File: src/components/card/product-card/ProductCard.jsx [hunk 1]
```diff
@@ -25,6 +33,45 @@ const ProductCard = ({ details, loading, cardIsOnPage }) => {
...
+      const config = {
+        headers: {
+          authorization: userState.userAuthToken,
+        },
+      };
+
+      if (isInWishList) {
+        await axios.delete(`/api/user/wishlist/${_id}`, config);
+
+        wishListDispatch({
```
Comment (vrx29) [inline src/components/card/product-card/ProductCard.jsx]: try to do api calls in try catch will be good while catching errors

**Review by vrx29 — APPROVED** [review]
Code looks really good and well maintained please go ahead and merge.

File: src/components/card/product-card/ProductCard.jsx (hunk 1 above)
Comment (hsnice16) [reply src/components/card/product-card/ProductCard.jsx]: Thanks for suggesting, I have done the changes.

**Comment by hsnice16** [comment]
Thanks you for reviewing

**Pull request merged**